fastfuels_sdk/exports.py

```diff
@@ -313,6 +326,336 @@ def export_zarr_to_fds(zroot: zarr.hierarchy.Group,
+            grass_min = [kwargs.get(key) for key in kwargs.keys() if key == "grass_min"][0]
+            litter_max = [kwargs.get(key) for key in kwargs.keys() if key == "litter_max"][0]
+            litter_min = [kwargs.get(key) for key in kwargs.keys() if key == "litter_min"][0]
+            grass_calibrated = _calibrate_maxmin(duet_grass,grass_max,grass_min)
+            litter_calibrated = _calibrate_maxmin(duet_litter,litter_max,litter_min)
+        # Combine grass and litter arrays
+        duet_calibrated = np.add(grass_calibrated,litter_calibrated)
+    elif fuel_types == ["none"]:
+        if keep_duet != "grass":
+            # Use max/min from SB40 to calibrate grass values
+            grass_max = np.max(rhof_arr[ftype_arr==1])
+            grass_arr = rhof_arr[ftype_arr==1]
+            grass_min = np.min(grass_arr[grass_arr>0])
+            grass_calibrated = _calibrate_maxmin(duet_grass,grass_max,grass_min)
+        if keep_duet != "litter":
+            litter_max = np.max(rhof_arr[ftype_arr==-1])
+            litter_arr = rhof_arr[ftype_arr==-1]
+            litter_min = np.min(litter_arr[litter_arr>0])
+            litter_calibrated = _calibrate_maxmin(duet_litter,litter_max,litter_min)
+        if keep_duet == "grass":
+            duet_calibrated = np.add(duet_grass,litter_calibrated)
+        elif keep_duet == "litter":
+            duet_calibrated = np.add(grass_calibrated,duet_litter)
+        else:
+            duet_calibrated = np.add(grass_calibrated,litter_calibrated)
+    
+    duet_calibrated = np.swapaxes(duet_calibrated,0,1) #need to to this or it doesn't match the uncalibrated duet outputs
+    _write_np_array_to_dat(duet_calibrated, "surface_rhof_calibrated.dat",
+                           output_dir, np.float32)
+    print("Calibration Complete")
+
+
+def replace_surface_fuels(zroot: zarr.hierarchy.Group,
+                          duet_dir: Path | str,
+                          quicfire_dir: Path | str,
+                          calibrated: bool) -> None:
+    """
+    Replace surface fuel bulk density in quicfire output
+    (from export_zarr_to_quicfire) with DUET output.
+
+    Parameters
+    ----------
+    zroot: zarr.hierarchy.Group
+        The root group of the zarr file.
+    duet_dir: Path | str
+        Directory where DUET .dat files are located.
+    quicfire_dir: Path | str
+        Directory where QUIO-Fire .dat files are located,
+        and to where updated .dat files are written to.
+    calibrated: bool
+        Whether the DUET surface fuel loading has been
+        calibrated using calibrate_duet.
+
+    Returns
+    -------
+    None
+        Modified bulk density array (treesrhof.dat) is written to the QUIC-Fire directory
+    """
+    duet_name = "surface_rhof_calibrated.dat" if calibrated else "surface_rhof.dat"
+    nx = zroot.attrs['nx']
+    ny = zroot.attrs['ny']
+    nz = zroot.attrs['nz']-2 #ask about this
```

Comment:
Good question. What are you running into here? I'm not sure that I've seen this where the DUET grid is a different height from the FastFuels zarr.